overlay/generic/smartdc/lib/smartos_prompt_config.sh

```diff
@@ -327,187 +772,158 @@ create_dump()
+		zfs create -o mountpoint=legacy ${OPTDS} || \
+		fatal "failed to create the opt dataset"
+		printf "%4s\n" "done"
+	fi
 
-    zpool_args=""
-
-    # When creating a mirrored pool, create a mirrored pair of devices out of
-    # every two disks.
-    if [[ ${profile} == "mirror" ]]; then
-        ii=0
-        for disk in ${disks}; do
-            if [[ $(( $ii % 2 )) -eq 0 ]]; then
-                  zpool_args="${zpool_args} ${profile}"
-            fi
-            zpool_args="${zpool_args} ${disk}"
-            ii=$(($ii + 1))
-        done
-    else
-        zpool_args="${profile} ${disks}"
-    fi
+	if ! echo $datasets | grep ${VARDS} > /dev/null; then
+		printf "%-56s" "Initializing var dataset... "
+		zfs create ${VARDS} || \
+		fatal "failed to create the var dataset"
+		chmod 755 /${VARDS}
+		cd /var
+		if ( ! find . -print | cpio -pdm /${VARDS} 2>/dev/null ); then
+			fatal "failed to initialize the var directory"
+		fi
 
-    zpool create -f ${pool} ${zpool_args} || \
-        fatal "failed to create pool ${pool}"
-    zfs set atime=off ${pool} || \
-        fatal "failed to set atime=off for pool ${pool}"
+		zfs set mountpoint=legacy ${VARDS}
+		printf "%4s\n" "done"
+	fi
 
-    printf "%4s\n" "done"
+	if ! echo $datasets | grep ${SWAPVOL} > /dev/null; then
+		printf "%-56s" "Creating swap zvol... "
+
+		#
+		# We cannot allow the swap size to be less than the size of
+		# DRAM, lest we run into the availrmem double accounting
+		# issue for locked anonymous memory that is backed by
+		# in-memory swap (which will severely and artificially limit
+		# VM tenancy).  We will therfore not create a swap device
+		# smaller than DRAM -- but we still allow for the
+		# configuration variable to account for actual consumed space
+		# by using it to set the refreservation on the swap volume
+		# if/when the specified size is smaller than DRAM.
+		#
+
+		size=${SYSINFO_MiB_of_Memory}
+		zfs create -V ${size}mb ${SWAPVOL} || fatal \
+		    "failed to create swap partition"
+
+		swap -a /dev/zvol/dsk/${SWAPVOL}
+	    fi
+	    printf "%4s\n" "done"
 }
-create_zpools()
+
+create_zpool()
 {
-  devs=$1
-
-  export SYS_ZPOOL="zones"
-  create_zpool "$devs"
-  sleep 5
-
-  svccfg -s svc:/system/smartdc/init setprop config/zpool="zones"
-  svccfg -s svc:/system/smartdc/init:default refresh
-
-  export CONFDS=${SYS_ZPOOL}/config
-  export COREDS=${SYS_ZPOOL}/cores
-  export OPTDS=${SYS_ZPOOL}/opt
-  export VARDS=${SYS_ZPOOL}/var
-  export USBKEYDS=${SYS_ZPOOL}/usbkey
-  export SWAPVOL=${SYS_ZPOOL}/swap
-
-  setup_datasets
-  #
-  # Since there may be more than one storage pool on the system, put a
-  # file with a certain name in the actual "system" pool.
-  #
-  touch /${SYS_ZPOOL}/.system_pool
+	layout=$1
+	disks=$2
+	pool=zones
+
+	# If the pool already exists, don't create it again.
+	if /usr/sbin/zpool list -H -o name $pool >/dev/null 2>/dev/null; then
+		printf "%-56s\n" "Pool '$pool' exists, skipping creation... "
```

Comment:
@jdwoody could you clarify what the problem is with this newer installer?
I would have thought that you would select "manual", import the pool, then close the shell and everything would be fine.
Or did "current install process" refer to the current version being shipped rather than the current version of this patch?